### PR TITLE
Remove unused custom CSS files and Sphinx directives

### PR DESCRIPTION
**Pull request by Eric-Arellano**
No one was using the custom directives `galleryitem`, `customgalleryitem`, `customcarditem`, and `includenodoc`. (I searched all Qiskit repos using ripgrep locally.)

After removing this, it's safe to remove `sphinx-gallery` because it's not used. That allows removing `gallery.css`, which is meant to override `sphinx-gallery`'s default styling.

This also removes `custom.css`. There is no `.toggle` element anymore.

### CODE REVIEW

File: docs/custom_directives.py [hunk 1]
```diff
@@ -10,19 +10,12 @@
...
 
-import sphinx_gallery
 from docutils import nodes
 from docutils.parsers.rst import Directive, directives
 from docutils.statemachine import StringList
 
-try:
-    FileNotFoundError
-except NameError:
-    FileNotFoundError = IOError
```
Comment (Eric-Arellano) [inline docs/custom_directives.py:24]: Legacy Python 2 code, I believe

**Comment by javabster** [comment]
what are we using for tutorials if we're not using sphinx-gallery? I always thought that was the package that was used to lay out all the boxes on the tutorials page, unless I'm mistaken?

**Comment by Eric-Arellano** [comment]
> what are we using for tutorials if we're not using sphinx-gallery? I always thought that was the package that was used to lay out all the boxes on the tutorials page, unless I'm mistaken?

Almost every project in Qiskit is using `nbsphinx`, which also has a tutorial gallery feature. For example, we use that in our example_docs/. 

Meanwhile, not one Qiskit project is using sphinx-gallery anymore. I suspect that you all migrated from sphinx-gallery to nbsphinx in the past.

**Review by javabster — DISMISSED** [review]
LGTM 🚀

**Review by javabster — DISMISSED** [review]

**Review by javabster — APPROVED** [review]

**Pull request merged**